### DISCUSSION TITLE
Recommended patterns for common usecases.

### DISCUSSION THREAD

**Discussion by Pipe-Runner** [motiondivision/motionone#31] [Support] [answered]
First of all, thanks a lot @mattgperry for this initiative. 
I am utilising this library in my personal [website](https://piperunner.in/). The code is in react and I know everyone would suggest using Framer for this usecase, but I like working with imperative libraries in non-production environments. Since Motion has a very small bundle size, it is a great fit for my project.
Now coming to the question at hand - I have a couple of common scenarios where I plan to use motion, and one of them is animating the menu drawer and the overlay underneath. It works as expected but I failed to recognise a couple of pitfalls.

Q. Getting rid of hidden DOM elements using `display: none` after they have been animated out of the screen or hidden via `opacity` in order to reduce memory footprint. I know `display` cannot be animated but the current flow kinds of makes the process of using `display` before and after animation seems like a disconnected activity. I am a bit new to advanced animations and I am not sure if using `display` to remove elements when not needed is a common thing to do. 
**Note**: Don't worry, the elements I am removing is in its own composite layer so reflow won't be a problem.

Q. I might digress a bit here. While animating the drawer, I don't want people to cancel the animation until it has finished. The obvious way of going about this is probably to retain some sort of an `isAnimating` state. But is there a more elegant way to approach this kind of problem? 

**Note**: Sorry if my questions are off topic. I don't intend to waste anyone's time. If the questions I am asking are too trivial, do mention that so that I can put in a bit more work before I ask questions next time.

**Comment by mattgperry** [comment, marked as answer]
For the first one, `visibility` can be animated as you'd expect it so I would recommend doing that. Especially if it's not in the layout flow. 

```javascript
animate({ opacity: 0, visibilty: "hidden" }) // Will hit visibility hidden on final frame
```


But even so

```javascript
const animation = animate()
animation.finished(() => { element.style.display = "none" })
```

Would work. 

As for blocking animation, ideally it would actually be interruptible which I know isn't always super convenient but will lead to a more fluid UX. Otherwise yeah you do need to store a `isAnimating` state separately, there's nothing really wrong with this, I do it quite often in Framer Motion.
